### PR TITLE
depends: qt 5.9.7

### PR DESCRIPTION
**Pull request by fanquake**
This PR upgrades `expat` and `qt` in depends. The intention is to upgrade Qt in master to the latest point release of the current Qt LTS. This change can then be back-ported to the 0.17 branch (wether it makes it into 0.17.1 or not). 

Then, sometime before the 0.18.0 release, we could move to using Qt 5.12+ in depends (which is also LTS). That discussion, as well as minimum supported Qt versions is in #13478.

### Qt 5.9.7
[Release announcement](https://blog.qt.io/blog/2018/10/23/qt-5-9-7-released/)
[Changelog](https://bugreports.qt.io/browse/QTBUG-70888?filter=20149)

### Expat 2.2.6
* Avoid doing arithmetic with NULL pointers in XML_GetBuffer
* Fix 2.2.5 regression with suspend-resume while parsing a document like <root/>

Full changelog [here](https://github.com/libexpat/libexpat/blob/R_2_2_6/expat/Changes)

https://github.com/bitcoin/bitcoin/pull/14849/commits/a46c8476e9598742e52944b6270b1854c8f500a2 disables a bunch of qt features we aren't currently using. This speeds up the qt depends build slightly (also decreases the size of the built `qt-5.9.7` tar by about 2%). The disabling is somewhat unintuitive, hence `[wip]` until after a travis run and gitian build.

### CODE REVIEW

**Comment by hebasto** [comment]
I've look through `./configure --help` and not sure if these options:
`-no-egl`
`-no-libudev`
`-no-openvg`
`-optimized-qmake`
are acceptable by Qt 5.9.7 `config`.

File: depends/packages/qt.mk [hunk 1]
```diff
@@ -69,9 +70,17 @@ $(package)_config_opts += -system-zlib
...
 $(package)_config_opts += -silent
 $(package)_config_opts += -v
+$(package)_config_opts += -no-feature-dial
+$(package)_config_opts += -no-feature-ftp
+$(package)_config_opts += -no-feature-lcdnumber
+$(package)_config_opts += -no-feature-pdf
 $(package)_config_opts += -no-feature-printer
 $(package)_config_opts += -no-feature-printdialog
 $(package)_config_opts += -no-feature-concurrent
+$(package)_config_opts += -no-feature-sql
```
Comment (hebasto) [inline depends/packages/qt.mk]: `./configure --list-features` output does not list `sql` feature.

**Review by hebasto — CHANGES_REQUESTED** [review]

**Comment by hebasto** [comment]
Some thoughts about disabled Qt features.
1. Qt build does not include [`QtPrintSupport`](https://doc.qt.io/qt-5.9/qtprintsupport-module.html) module; therefore, it is redundant to mention:
```
$(package)_config_opts += -no-feature-printer
$(package)_config_opts += -no-feature-printdialog
```

2. Qt build does not include [`QtConcurrent`](https://doc.qt.io/qt-5.9/qtconcurrent-module.html) module; therefore, it is redundant to mention:
```
$(package)_config_opts += -no-feature-concurrent
```

3. Also these features can be disabled:
```
$(package)_config_opts += -no-feature-udpsocket
$(package)_config_opts += -no-feature-syntaxhighlighter

```

File: depends/packages/qt.mk (hunk 1 above)
Comment (fanquake) [reply depends/packages/qt.mk]: Passing `-no-feature-sql` disables the Qt sql module in its entirety, see the output in the configure summary:
```
Qt modules and options:
  Qt Network ............................. yes
  Qt Sql ................................. no
  Qt Testlib ............................. yes
```

**Comment by fanquake** [comment]
@hebasto Thanks for your thoughts:

> $(package)_config_opts += -no-feature-printer
$(package)_config_opts += -no-feature-printdialog

These were specifically introduced to [fix the depends build on macOS](https://github.com/bitcoin/bitcoin/pull/9574). I'd rather leave them in place for now. Similar thoughts for `-no-egl -no-libudev -no-openvg -optimized-qmake`. Maybe once we move to Qt 6 we can review the configure flags we are using. However for now I'd rather only remove extra unused features/reduce compile times etc, than inadvertently break any depends/gitian builds by removing a potential unused flag.

> $(package)_config_opts += -no-feature-concurrent

We can leave this module disabled, and not do any extra compilation/work during the Qt compile. i.e
```
Qt modules and options:
  Qt Concurrent .......................... no
```

> $(package)_config_opts += -no-feature-udpsocket
$(package)_config_opts += -no-feature-syntaxhighlighter

I've added these + `-no-feature-statemachine`  to the PR.

**Comment by hebasto** [comment]
@fanquake You are right. Qt docs confused me a little :)
utACK 311498de728bd584d51865aa62e06091235baab5

**Comment by jonasschnelli** [comment]
Successful gitian build:
https://bitcoin.jonasschnelli.ch/build/907

Tested on OSX:
bildschirmfoto 2018-12-09 um 13 55 17

**Comment by fanquake** [comment]
Rebased on master and removed [wip].

**Comment by laanwj** [comment]
utACK a46c8476e9598742e52944b6270b1854c8f500a2

**Pull request merged**